src/frontend/src/common/components/library/data_table/index.tsx

```diff
@@ -112,13 +151,45 @@ export const DataTable: FunctionComponent<Props> = ({
   handleHeaderCheckboxClick,
   handleRowCheckboxClick,
   isHeaderChecked,
+  setIsHeaderChecked,
   isHeaderIndeterminant,
+  setHeaderIndeterminant,
 }: Props) => {
   const [state, dispatch] = useReducer(reducer, {
     ascending: false,
     sortKey: defaultSortKey,
   });
 
+  useEffect(() => {
```

Comment:
I'm a little confused why this all needs to happen in DataTable instead of in the parent component. It seems like `data`, and the state modifiers are all passed into this function. Are you able to do all this calculation in the parent component and then pass it down as a simplified/reduced prop instead of passing down all the info and functions needed to calculate it?

Reply:
so this needs to happen in data table because that's where the table rows are being rendered based on filtering (and we need  access to this subset of data to tell which state the header should be (indeterminate or checked)), data in the subview is a list of all available rows (prior to any filtering that's done).

Reply:
i reorganized this part to be more straightforwards